ApplicationCode/Commands/ToggleCommands/RicMoveItemsToTopFeature.cpp

```diff
@@ -0,0 +1,156 @@
+/////////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (C) 2020-     Equinor ASA
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "RicMoveItemsToTopFeature.h"
+
+#include "RiaGuiApplication.h"
+#include "RicToggleItemsFeatureImpl.h"
+#include "RiuMainWindow.h"
+
+#include "cafPdmFieldReorderCapability.h"
+#include "cafPdmObject.h"
+#include "cafPdmObjectHandle.h"
+#include "cafPdmPtrArrayFieldHandle.h"
+#include "cafPdmUiObjectHandle.h"
+#include "cafPdmUiTreeView.h"
+#include "cafSelectionManager.h"
+
+#include <QAction>
+
+CAF_CMD_SOURCE_INIT( RicMoveItemsToTopFeature, "RicMoveItemsToTopFeature" );
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RicMoveItemsToTopFeature::isCommandEnabled()
+{
+    using namespace caf;
+
+    std::vector<caf::PdmUiItem*> selectedItems;
+    caf::SelectionManager::instance()->selectedItems( selectedItems );
+
+    if ( !selectedItems.empty() )
+    {
+        PdmUiObjectHandle* uiObjectHandle = dynamic_cast<PdmUiObjectHandle*>( selectedItems[0] );
+
+        if ( uiObjectHandle )
+        {
+            PdmObjectHandle* pdmObject = uiObjectHandle->objectHandle();
+            if ( pdmObject )
+            {
+                auto reorderability = PdmFieldReorderCapability::reorderCapabilityOfParentContainer( pdmObject );
+                if ( reorderability )
+                {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RicMoveItemsToTopFeature::onActionTriggered( bool isChecked )
+{
+    using namespace caf;
+
+    std::vector<caf::PdmUiItem*> selectedItems;
+    caf::SelectionManager::instance()->selectedItems( selectedItems );
+
+    if ( !selectedItems.empty() )
+    {
+        PdmUiObjectHandle* uiObjectHandle = dynamic_cast<PdmUiObjectHandle*>( selectedItems[0] );
+
```

Comment:
Dette ble en *veldig* dyp if-struktur. Kanskje noen continue eller tidlig return istedenfor? Eventuelt en hjelpefunksjon for å hente ut et valgt PdmObject og parent arrayField?